lib/ruby_lsp/requests/support/rubocop_diagnostics_runner.rb

```diff
@@ -32,14 +32,15 @@ def initialize
 
         sig { params(uri: String, document: Document).returns(T::Array[Support::RuboCopDiagnostic]) }
         def run(uri, document)
+          @diagnostics.clear
```

Comment:
Now that we moved to a singleton runner for diagnostics, we need to clear the array otherwise it may already be populated from previous runs. I noticed this because the tests were flaky.